src/lib/raft/raft.c

```diff
@@ -1041,8 +1144,7 @@ raft_cfg_death_timeout(struct raft *raft, double timeout)
 		return;
 
 	raft->death_timeout = timeout;
-	if (raft->state != RAFT_STATE_FOLLOWER || !raft->is_candidate ||
-	    raft->leader == 0)
+	if (raft->state != RAFT_STATE_FOLLOWER || !raft_is_leader_seen(raft))
```

Comment:
It might be important to test everything, this change didn't break the unit tests somewhy:
```Diff
@@ -1144,7 +1144,7 @@ raft_cfg_death_timeout(struct raft *raft, double timeout)
 		return;
 
 	raft->death_timeout = timeout;
-	if (raft->state != RAFT_STATE_FOLLOWER || !raft_is_leader_seen(raft))
+	if (raft->state != RAFT_STATE_FOLLOWER || raft->leader == 0)
 		return;
```

Reply:
Covered. Thanks for pointing this out!